### PR TITLE
WorkInProgress - Remove UIKit & cleanup files

### PR DESCRIPTION
**Pull request by scramb**
# Work-in-progress

## Description

Added vuetify to files-app and adjusted core to use vuetify - no longer depending on UiKit.
Cleanup on store, and app-loading. While removing UiKit removed less-loader. 

## Related Issue

- refs #1 
- refs #279 
- refs #298 

## Motivation and Context

I tried to remove UiKit and replace it with vuetify, this solves the problem with themability, rtl-support and touch support. Would really love to get feedback on this while working on it.

## How Has This Been Tested?

- WIP not tested yet.

## Screenshots (if appropriate):
2018-11-16-113221_1919x904_scrot
2018-11-22-111654_1906x901_scrot
2018-11-22-111631_1903x863_scrot

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Notification/Snackbar to queue all notifications
- [x] add files functions like createFolder
- [x] adjust File-Details.vue in apps/files
- [x] Side bar is not opening - let's chat on how to activate it ... see #149
- [x] new folder - can enter name but no action on [enter]
- [x] file filter not working

### CODE REVIEW

**Comment by tempelgogo** [comment]
The vuetify Snackbar is unfortunately implemented as a component.
I've written a NotificationBar component wrapping up the vuetify Snackbar, reading a new notification queue from vuex store.
I do also provide a global vuex action to add a new Notification, analogous to the global $uikit.notification() method.

Some questions are open before i can release the code:
* respecting the Material Design Guidelines, multiple Toast/Snackbar messages are not allowed.
What is the intended behaviour, if multiple notifications are thrown ?
    * ignore the first, show last
    * ignore the last, show first
    * enqueue them, show them in sequence respecting their timeout
    * ignore the MD guidelines, rewrite CSS & allow multiple notifications ?
* is a global reachable addNotification action ok; Or should i create some global "PhoenixUI" object, which we can expose for apps to interact with Phoenix Core ?

I personally prefer to enqueue them, and show them consecutively

**Comment by scramb** [comment]
@tempelgogo thanks for your feedback - i would prefer to show them consecutively aswell!

**Comment by DeepDiver1975** [comment]
>     * respecting the Material Design Guidelines, multiple Toast/Snackbar messages are not allowed.

a good question for the ux workshop - please keep a note. For now we choose the less effort implementation.

File: apps/files/src/components/Files-App.vue [hunk 1]
```diff
@@ -1,209 +1,225 @@
...
-								// --- Filedate ------------------
-								td(class="uk-visible@s").uk-text-nowrap.uk-table-shrink
-									time.uk-text-meta {{ file.mdate | formDateFromNow }}
-
-				aside.uk-width-medium.uk-background-default.uk-padding-small(v-show="selected.length > 0", class="uk-width-large@l uk-padding@l").uk-animation-slide-right-small
-					FileDetails(@reset="resetFileSelection")
+<template>
+	<v-container id="files-app" fluid class="pa-0">
+		<v-toolbar class="elevation-1">
+ 			<v-btn v-if="!createFolder" @click="createFolder = !createFolder" flat><v-icon large>create_new_folder</v-icon></v-btn>
```
Comment (DeepDiver1975) [inline apps/files/src/components/Files-App.vue]: can we wrap lines at ~80 chars? This allows easier review on github - THX

File: apps/files/src/components/Files-App.vue [hunk 2]
```diff
@@ -1,209 +1,225 @@
...
+			<v-btn v-if="createFolder" @click="addNewFolder(newFolderName)" flat><v-icon large>add</v-icon></v-btn>
+			<v-flex v-if="createFolder" xs2>
+				<v-text-field
+					placeholder="New folder Name"
```
Comment (DeepDiver1975) [inline apps/files/src/components/Files-App.vue]: translations missing - right?

File: src/components/Menu.vue [hunk 3]
```diff
@@ -1,63 +1,136 @@
...
+
+            <v-divider ></v-divider>
+
+            <v-list-tile
+              @click="logout()">
+              <v-list-tile-action>
+                <v-icon>exit_to_app</v-icon>
+              </v-list-tile-action>
+              <v-list-tile-content>
+                <v-list-tile-title>Exit ownCloud</v-list-tile-title>
```
Comment (DeepDiver1975) [inline src/components/Menu.vue]: translation?

Comment (DeepDiver1975) [reply src/components/Menu.vue]: brand shall be taken from theme

File: src/components/Menu.vue [hunk 4]
```diff
@@ -1,63 +1,136 @@
...
 </template>
 
 <script>
+import { mapActions, mapGetters } from 'vuex'
 export default {
+  data () {
+    return {
+      menuData: {
+        personal: {
+          title: 'Personal',
```
Comment (DeepDiver1975) [inline src/components/Menu.vue]: translations as well .....

File: apps/files/src/components/Files-App.vue (hunk 2 above)
Comment (scramb) [reply apps/files/src/components/Files-App.vue]: yes - will bring them in them in one of the next commits

**Comment by DeepDiver1975** [comment]
- [ ] Side bar is not opening - let's chat on how to activate it ... see #149

**Comment by scramb** [comment]
>     * [ ]  Side bar is not opening - let's chat on how to activate it ... see #149
> 
>     * [ ]  new folder - can enter name but no action on [enter]
> 
>     * [ ]  file filter not working

added those to the To-Do's -  i'm refcatoring while re-enabling existing functions and did not came through to put the up again... pug took me a long time...

**Comment by DeepDiver1975** [comment]
> pug took me a long time...

"Ceterum censeo Pugem esse delendam"

File: src/components/Menu.vue [hunk 5]
```diff
@@ -58,7 +58,7 @@
                 <v-icon>exit_to_app</v-icon>
               </v-list-tile-action>
               <v-list-tile-content>
-                <v-list-tile-title>Exit ownCloud</v-list-tile-title>
+                <v-list-tile-title>Exit {{ configuration.theme.general.name }}</v-list-tile-title>
```
Comment (DeepDiver1975) [inline src/components/Menu.vue]: missing v-translate ???

File: src/components/Menu.vue (hunk 4 above)
Comment (scramb) [reply src/components/Menu.vue]:
```suggestion
            <v-list-group
                      v-for="item in menuData"
                      v-model="item.active"
                      :key="item.title"
                      no-action
                    >

                      <v-list-tile slot="activator">
                        <v-list-tile-action>
                          <v-icon>{{ item.icon }}</v-icon>
                        </v-list-tile-action>
                        <v-list-tile-content>
                          <v-list-tile-title v-translate>{{ item.title }}</v-list-tile-title>
                        </v-list-tile-content>
                      </v-list-tile>
                      <v-list-tile
                        v-for="subItem in item.subItems"
                        :key="subItem.title"
                        @click="notImplemented()"
                      >
                        <v-list-tile-content>
                          <v-list-tile-title v-translate>{{ subItem.title }}</v-list-tile-title>
                        </v-list-tile-content>

                        <v-list-tile-action>
                          <v-icon>{{ subItem.action }}</v-icon>
                        </v-list-tile-action>
                      </v-list-tile>
                    </v-list-group>
```
does this allready

**Comment by DeepDiver1975** [comment]
build is :boom: 

```
ERROR in ./src/components/File-Details-Buttonrow.vue?vue&type=template&id=8b7dbf44&lang=pug& (./node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!./node_modules/vue-loader/lib??vue-loader-options!./src/components/File-Details-Buttonrow.vue?vue&type=template&id=8b7dbf44&lang=pug&)
Module Error (from ./node_modules/vue-loader/lib/loaders/templateLoader.js):
(Emitted value instead of an instance of Error) 
  Error compiling template:
  
  
  // --- pug mixin ----------------------------------------------------
  
  mixin icon(a)
  	i.material-icons(class=a.class)= a.icon
  
  mixin iconButton(a)
  	button.uk-button(class='uk-button-' + a.buttonType, class=a.class, type="button", @click=a.click).uk-flex.uk-flex-middle
  		if a.text
  			+icon({icon: a.icon, class: 'uk-margin-small-right'})
  			span= a.text
  		else
  			+icon({icon: a.icon})

**Pull request merged**